### DISCUSSION TITLE
[F] Return null/undefined instead of throwing an error when getting a column by an inexistent key

### DISCUSSION THREAD

**Discussion by sor-geison** [exceljs/exceljs#2457] [Ideas & Proposals]
## 🚀 Feature Proposal

If you call for example `worksheet.getColumn('client')` and column with key 'client' doesn't exists it throws the error:

`Out of bounds. Excel supports columns from 1 to 16384`

## Motivation

I have columns that are created dynamically and it's annoying to check for the condition that determines if they are created instead of checking if they themselves exist when parsing the numberFormat on each cell of said column.

## Example

Please provide an example for how this feature would be used.

`console.log(worksheet.getColumn('client'))`

would print `undefined`

not throw the error:

`{
 name: 'Error',
message: 'Out of bounds. Excel supports columns from 1 to 16384',
level: 'err'
}
`

**Comment by PrimulaX** [comment]
This one would be good.

**Comment by akomm** [comment]
It's just crazy. When you have column keys, there is no way to *try* get a cell by that key via row, you need to lookup via sheet to resolve it to letter and row number and then try find it. You can do it but all upside-down. Or it has been hidden very well and left undocumented.

![grafik](https://github.com/exceljs/exceljs/assets/2409779/3c9b71ae-ae69-4b14-b897-26f2064259d9)
